Utils/Dataflow/pyDKB/dataflow/stage/ProcessorStage.py

```diff
@@ -115,53 +115,61 @@ def defaultArguments(self):
         """ Default parser configuration. """
         super(ProcessorStage, self).defaultArguments()
         self.add_argument('input_files', type=str, nargs='*',
-                          help=u'Source data file.',
+                          help=u'source data file\n'
+                          'NOTE: required in (f)ile, '
+                          'ignored in other modes',
                           metavar=u'FILE'
                           )
         self.add_argument('-s', '--source', action='store', type=str,
-                          nargs='?',
-                          help=u'Where to get data from: '
-                                'local (f)iles, (s)tdin, '
-                                '(h)dfs (same as --hdfs).',
+                          help=u'where to get data from:\n'
+                          '    f -- local files\n'
+                          '    s -- stdin\n'
+                          '    h -- hdfs files',
                           default='f',
-                          const='f',
                           choices=['f', 's', 'h'],
                           dest='source'
                           )
         self.add_argument('-i', '--input-dir', action='store', type=str,
-                          nargs='?',
-                          help=u'Base directory in local file system '
-                                'or in HDFS (for relative FILE names). '
-                                'If no FILE specified, all files from the '
-                                'directory will be taken.',
-                          default='',
-                          const='',
+                          help=u'base directory in local file system '
+                          'or in HDFS (for relative FILE names). '
+                          'If no FILE specified, all files with '
```

Comment:
Is specified? Was specified?

Reply:
Both. Any.
It seems to be quite common to say “no file specified” in error messages, so it is just stuck in my mind.
It is something about “omission of ‘to be’ in passive” or something like this.
[Here](
https://forum.wordreference.com/threads/passive-voice-omission-of-be.2554621/#post-12874978) is one of the first hits of what people say about it.

Reply:
I've read the forum thread, but I still believe that something should be changed here. I agree with you - short forms "No X specified", "missing X" and some others are pretty common in error messages. However, here we are talking about a complete sentence, not "something to say", as the thread defines it. There is a possibility to make it look normal, but that means changing the whole message and all other messages to similar style, i.e. "Source directory. If no FILE specified, take all files from X which match input extension".

Reply:
Reworded.